### ISSUE TITLE
Improve logging for msal4j

### ISSUE THREAD

**Issue by siddhijain** [AzureAD/microsoft-authentication-library-for-java#499]
Recently, one of the 1P customers faced issue while following B2C docs and reported following things - 

I have been over the ref docs https://aka.ms/msal4j-b2c-known-issues multiple times and it did not help me find the solution:
1) the docs is not clear about what scnearios the AT was not returned and there is no way for me to tell if the AT was returned without seeing some kind of http request/response
2) the suggestion to use client id does not work and even throw an error so either that docs is outdated or something has changed
) The root cause of AT missing due to consent is not very obvious in this case (Ijust discover that by accidence) since B2C does not throw any error nor does it present a consent screen (maybe this is bug in B2C service?).  It just simply return 200 missing token.  This was verified in postman.
 
Bottom line is that neither error message nor reference docs are helpful in resolving this problem.

This issue is to make improve logging to reflect correct error messages.

**Comment by siddhijain** [comment]
Moving this to the backlog as it is not a customer ask.

https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20JAVA/Features/?showParents=true&workitem=2441675
